src/submission/logic.py

```diff
@@ -473,3 +473,120 @@ def add_author_from_search(search_term, request, article):
         )
 
     return author
+
+
+def add_new_author_from_form(request, article):
+    new_author_form = EditFrozenAuthor(request.POST)
+    author = None
+
+    frozen_email = request.POST.get("frozen_email")
+    if frozen_email:
+        try:
+            author = models.FrozenAuthor.objects.get(
+                Q(article=article) &
+                (Q(frozen_email=frozen_email) |
+                Q(author__username__iexact=frozen_email))
+            )
+        except models.FrozenAuthor.DoesNotExist:
+            pass
+
+    if not author and new_author_form.is_valid():
+        author = new_author_form.save()
+        author.article = article
+        author.order = article.next_frozen_author_order()
+        author.save()
+    else:
+        messages.add_message(
+            request, messages.WARNING,
+            _('Could not add the author manually.'),
+        )
+
+    if author:
+        messages.add_message(
+            request, messages.SUCCESS,
+            _('%(author_name)s (%(email)s) added to the article.')
+                % {
+                    "author_name": author.full_name(),
+                    "email": author.email
+                },
+        )
+    return author
+
+
+def save_correspondence_author(request, article):
+    account = get_object_or_404(
+        core_models.Account,
+        pk=request.POST.get('corr_author', None),
+        frozenauthor__article=article,
+    )
+    article.correspondence_author = account
+    article.save()
+    messages.add_message(
+        request,
+        messages.SUCCESS,
+        _('%(author_name)s (%(email)s) made correspondence author.')
+            % {
+                "author_name": account.full_name(),
+                "email": account.email
+            },
+    )
+    author = account.frozen_author(article)
+    return author
+
+
+def get_credit_form(request, author):
+    use_credit = setting_handler.get_setting(
+        "general",
+        "use_credit",
+        journal=request.journal,
+    ).processed_value
+    if use_credit:
+        return CreditRecordForm(
+            frozen_author=author,
+        )
+    else:
+        return None
+
+
+def add_credit_role(request, article):
+    author = get_object_or_404(
+        models.FrozenAuthor,
+        pk=int(request.POST.get('author_pk')),
```

Comment:
if we are not using a Django form, we should take care of sanitizing this input before we hand it over to the ORM and raise validation errors as required (rather than raise unexpected typ errors)